### PR TITLE
Inheritance

### PR DESCRIPTION
**Pull request by adamsjr8576**
@Jessmaxim303 @Dustin-Mikusko @khalidwilliams @brittanystoroz 

**What does this PR do??**

 - This PR creates a parent class of userFitness that is then inherited by the Sleep, Hydration and Activity classes. This is meant to DRY up the code base and take all repeating functions into the parent class. It also made it possible to refactor some of the existing methods on the child classes in regards to how they receive the user data pertinent to that class.

- This PR updates the tests accordingly including making a new test file for the userFitness class.

- This PR updates the script.js file according to the changes made on each class and does not affect the ux of the site at all.

**Where should the reviewer start?**

 - The viewer should start by pulling down and looking at the userFitness.js file, which is the parent of the Sleep, Hydration and Activity classes. Then look at how that affects those child classes, specifically how they interact with one another. It is helpful to use the debugger to see how JS is stepping through these files (instantiating the child classes), which helps to show how it is working and connected.

**How should this be manually tested?**

- Pull down this branch, run both npm run test and npm run start to make sure your local machine passes the tests, and the localhost shows the script.js updates.

**What are the relevant tickets?** #25

### CODE REVIEW

File: src/UserFitness.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
+class UserFitness {
+  constructor(data, user) {
+    this.userData = this.findUser(data, user.id);
+  }
+
```
Comment (Dustin-Mikusko) [inline src/UserFitness.js]: This is an excellent way to instantiate a specific user each time a new sub-class is made. Great work on this. This, along with refactoring each sub-class is going to clean up the code so much!

File: src/Sleep.js [hunk 2]
```diff
@@ -1,58 +1,40 @@
...
-  }
+import UserFitness from "./UserFitness";
 
-  returnWeek(week) {
-    var specificUser = this.findUser()
-    return [...specificUser].splice(-7 * week, 7).map(day => day.date);
+class Sleep extends UserFitness {
+  constructor(sleepData, user) {
+    super(sleepData, user);
   }
```
Comment (Dustin-Mikusko) [inline src/Sleep.js]: This is good stuff. Each sub-class gets only the info it needs from the parent class.

**Comment by Dustin-Mikusko** [comment]
This PR closes #25

File: src/UserFitness.js [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+  findUser(data, id) {
+    return data.filter(user => user.userID === id);
+  }
+
+  returnWeekOfData(week) {
```
Comment (khalidwilliams) [inline src/UserFitness.js]: The parameter could be a little more specific here -- maybe `numberOfWeeks`?

**Pull request merged**